### PR TITLE
Take focus off of types, concentrate on Elm Architecture in first example

### PR DESCRIPTION
**Pull request by willnwhite**
See https://github.com/evancz/guide.elm-lang.org/issues/10

### CODE REVIEW

File: architecture/user_input/buttons.md [hunk 1]
```diff
@@ -5,29 +5,24 @@
...
 ```elm
-import Html exposing (Html, button, div, text)
-import Html.App as Html
+import Html exposing (div, button, text)
+import Html.App exposing (beginnerProgram)
 import Html.Events exposing (onClick)
 
 
 main =
-  Html.beginnerProgram { model = model, view = view, update = update }
```
Comment (willnwhite) [inline architecture/user_input/buttons.md]: looks redundant, at least compared to initialising the model there

File: architecture/user_input/buttons.md [hunk 2]
```diff
@@ -5,29 +5,24 @@
...
-model =
-  0
+model = 0
 
 
 -- UPDATE
 
 type Msg = Increment | Decrement
 
-update : Msg -> Model -> Model
```
Comment (willnwhite) [inline architecture/user_input/buttons.md]: two arrows!?

File: architecture/user_input/buttons.md [hunk 3]
```diff
@@ -50,12 +44,10 @@ view model =
 
 That's everything!
 
-> **Note:** This section has `type` and `type alias` declarations. You can read all about these in the upcoming section on [types](http://guide.elm-lang.org/types/index.html). You do not *need* to deeply understand that stuff now, but you are free to jump ahead if it helps.
-
 When writing this program from scratch, I always start by taking a guess at the model. To make a counter, we at least need to keep track of a number that is going up and down. So let's just start with that!
 
 ```elm
-type alias Model = Int
```
Comment (willnwhite) [inline architecture/user_input/buttons.md]: Despite Note above to the contrary.

File: architecture/user_input/buttons.md [hunk 4]
```diff
@@ -5,29 +5,24 @@
...
+  beginnerProgram { model = 0, view = view, update = update }
 
 
 -- MODEL
 
-type alias Model = Int
-
-model : Model
-model =
-  0
```
Comment (willnwhite) [inline architecture/user_input/buttons.md]: If I didn't know about types yet: `model = 0` and `Model = Int` I might think "two models!?"

**Comment by willnwhite** [comment]
I've come up with a nicer way to introduce type signatures, which includes introducing compiler error messages and the package documentation. Perhaps we could talk about it if you approve of the basic idea here of simplifying the first example's code.

File: architecture/user_input/buttons.md [hunk 5]
```diff
@@ -5,29 +5,24 @@
...
+model = 0
 
 
 -- UPDATE
 
 type Msg = Increment | Decrement
```
Comment (willnwhite) [inline architecture/user_input/buttons.md]: The only reference to types still left. Provides an enticing curiosity only.

**Comment by evancz** [comment]
Put comments in #10. I fully agree with what you are saying, but just removing them isn't a fix without structural changes to the pacing and focus of other sections. So it's not ideal now, but I'd rather focus on fixing it in a coherent way.

**Pull request closed without merging**